src/main/kotlin/it/pagopa/ecommerce/helpdesk/services/PmService.kt

```diff
@@ -19,69 +21,34 @@ class PmService(@Autowired val connectionFactory: ConnectionFactory) {
         pageSize: Int,
         pmSearchTransactionRequestDto: PmSearchTransactionRequestDto
     ): Mono<SearchTransactionResponseDto> {
-        logger.info("[helpDesk pm service] searchTransaction method")
-
-        return Flux.usingWhen(
-                connectionFactory.create(),
-                { connection ->
-                    Flux.from(
-                            connection
-                                .createStatement("SELECT 'Hello, Oracle' FROM sys.dual")
-                                .execute()
+        logger.info(
+            "[helpDesk pm service] searchTransaction method, search type: ${pmSearchTransactionRequestDto.type}"
+        )
+        return pmTransactionDataProvider.totalRecordCount(pmSearchTransactionRequestDto).flatMap {
+            totalCount ->
+            if (totalCount > 0) {
```

Comment:
What if the user has no records for the requested search params? (i.e. an RPT id that was never activated)

Reply:
In the single datasource service implementation we return a 404 not found error to the caller.

In another PR will be implemented the two datasource merge solution that will perform pagination against eCommerce first and PM then and that will return 404 iff no record in found in both DB